### PR TITLE
Tiny fix trtllm_fp8_per_tensor_scale_moe_wrapper router_logits dtype

### PR DESCRIPTION
**Pull request by Qiaolin-Yu**
## Motivation

https://github.com/flashinfer-ai/flashinfer/blob/fe0539318dcc31c76a33a7ed2ab0ee3c94fe6bad/csrc/trtllm_fused_moe_kernel_launcher.cu#L1789

the dtype of router_logits should be float32 for deepseek routing method
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review and Merge Process

1. Ping Merge Oncalls to start the process. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - Common commands include `/tag-and-rerun-ci`, `/tag-run-ci-label`, `/rerun-failed-ci`
4. After green CI and required approvals, ask Merge Oncalls or people with Write permission to merge the PR.

### CODE REVIEW

**Review by b8zhong — APPROVED** [review]

File: python/sglang/srt/layers/moe/moe_runner/flashinfer_trtllm.py [hunk 1]
```diff
@@ -470,8 +470,15 @@ def fused_experts_none_to_flashinfer_trtllm_fp8(
         # Move kernel call outside context manager to avoid graph breaks
         # during torch.compile for piecewise cuda graph.
         # Use custom op wrapper for torch.compile compatibility.
+
+        # The DeepSeekV3 routing method requires float32 router logits.
```
Comment (nvpohanh) [inline python/sglang/srt/layers/moe/moe_runner/flashinfer_trtllm.py:474]: @leejnau @trevor-m is this true? If so, why didn't we run into issues before?

Comment (nvpohanh) [reply python/sglang/srt/layers/moe/moe_runner/flashinfer_trtllm.py:474]: Maybe will be fixed by https://github.com/flashinfer-ai/flashinfer/pull/2993 ?

Comment (trevor-m) [reply python/sglang/srt/layers/moe/moe_runner/flashinfer_trtllm.py:474]: The path for block scale had this fix already, I think we didn't use per tensor scaling before?

Comment (nvpohanh) [reply python/sglang/srt/layers/moe/moe_runner/flashinfer_trtllm.py:474]: got it. we have never run DSV3/R1 with per-tensor FP8 before.
